### PR TITLE
Feature 130/Add network connection checking for resource fetch errors

### PR DESCRIPTION
**Pull request by PhotoNomad0**
# Describe what your pull request addresses

- [ ] Improvements to ErrorPopup for more flexibility
- [ ] tweaks to ResourceCard to handle onResourceError() callbacks and check for network disconnects.

## Test Instructions
- note that there are differences in network behavior among browsers.  Firefox may show long delays before showing the network error, but chrome is often immediate.
- test with https://deploy-preview-207--gateway-edit.netlify.app/
- test network error handling while loading trans helps repos:
  - [ ] set org to `test_org`, and `en`, set to tit 1:4, all panes should be showing content
  - [ ] change language to `ru` (should see everything but tQ since it is not in tsv)
  - test network connection error when loading another book:
    - [ ] disconnect from network and change book to `jon`, 
    - [ ] after delay should see popup of network connection error

  Screen Shot 2021-05-20 at 8 59 36 AM

    - [ ] reconnect to network and click cancel
    - [ ] screens should load, but should see this because there are bad characters in support reference, but since network is connected should not see network disconnected popup:

    Screen Shot 2021-05-20 at 8 45 45 AM  

  - [ ] change tN to 2 of 5 and tA article should load
- testing network error handling loading helps articles
  - [ ] you should still be at jon 1:1. Disconnect from network
  - [ ] change vs to 17 and after long pause should eventually see network error:

  Screen Shot 2021-05-20 at 8 59 36 AM

  - [ ] reconnect to network and click retry.  Page should reload

- test network error handling while clicking links:
  - [ ] change ref to `tit 1:3`, change tN to 2 of 6.
  - [ ] disconnect from network and click this link on tA:

  Screen Shot 2021-05-20 at 9 08 53 AM

  - [ ] after a delay, should see a network error popup

  Screen Shot 2021-05-20 at 8 59 36 AM

  - [ ] reconnect network, click cancel
  - [ ] change ref to `tit 1:3`, change tN to 2 of 6.
  - [ ] click tA link again and it should load this time

# Reviewable: https://reviewable.io/reviews/unfoldingWord/gateway-edit/207#-

---
This change is [Reviewable](https://reviewable.io/reviews/unfoldingword/gateway-edit/207)

### CODE REVIEW

File: src/utils/resources.js [hunk 1]
```diff
@@ -25,6 +25,8 @@ export async function getResource({
 
   let resource
 
+  console.log(`getResource():`, {owner, languageId, resourceId, branch, resourceLink})
```
Comment (mannycolon) [inline src/utils/resources.js:28]: Do we need the console log?

**Review by PhotoNomad0 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/unfoldingword/gateway-edit/207)* status: 0 of 8 files reviewed, 1 unresolved discussion (waiting on @mannycolon)

___
*[src/utils/resources.js, line 28 at r1](https://reviewable.io/reviews/unfoldingword/gateway-edit/207#-MaFICW7RVpkmPW2oiQb-r1-28:-MaIh14D4gaIQKrirhX3:bss4a2b) ([raw file](https://github.com/unfoldingword/gateway-edit/blob/debec68f43930e21cc3723c26078f5397c06f798/src/utils/resources.js#L28)):*
<details><summary><i>Previously, mannycolon (Manny Colon) wrote…</i></summary><blockquote>

Do we need the console log?
</blockquote></details>

Removed.

File: src/utils/resources.js (hunk 1 above)
Comment (PhotoNomad0) [reply src/utils/resources.js]: REmoved

**Review by mannycolon — APPROVED** [review]
LGTM

**Pull request merged**